core/auth/src/main/java/software/amazon/awssdk/auth/credentials/InstanceProfileCredentialsProvider.java

```diff
@@ -69,56 +105,91 @@ public static InstanceProfileCredentialsProvider create() {
+            this.cachedCredentials = credentials;
+        } catch (RuntimeException | IOException e) {
+            credentials = this.cachedCredentials;
+
+            if (credentials != null) {
+                credentials.getExpiration().ifPresent(expiration -> {
+                    // Choose whether to report this failure at the debug or warn level based on how much time is left on the
+                    // credentials before expiration.
+                    Supplier<String> errorMessage = () -> "Failure encountered when attempting to refresh credentials from IMDS.";
+                    Instant fifteenMinutesFromNow = Instant.now().plus(15, MINUTES);
+                    if (expiration.isBefore(fifteenMinutesFromNow)) {
+                        log.warn(errorMessage, e);
+                    } else {
+                        log.debug(errorMessage, e);
+                    }
+                });
+            } else {
+                throw SdkClientException.create("Failed to load credentials from IMDS.", e);
+            }
+        }
 
-            boolean is400ServiceException = e instanceof SdkServiceException
-                    && ((SdkServiceException) e).statusCode() == 400;
+        return RefreshResult.builder(credentials.getAwsCredentials())
+                            .staleTime(null) // Allow use of expired credentials - they may still work
+                            .prefetchTime(prefetchTime(credentials.getExpiration().orElse(null)))
+                            .build();
+    }
 
-            // metadata resolution must not continue to the token-less flow for a 400
-            if (is400ServiceException) {
-                throw SdkClientException.builder()
-                        .message("Unable to fetch metadata token")
-                        .cause(e)
-                        .build();
-            }
+    private boolean isLocalCredentialLoadingDisabled() {
+        return SdkSystemSetting.AWS_EC2_METADATA_DISABLED.getBooleanValueOrThrow();
+    }
 
-            return null;
+    private Instant prefetchTime(Instant expiration) {
+        Instant now = Instant.now();
+        Instant oneHourFromNow = now.plus(1, HOURS);
+
+        // If expiration time is infinite or farther out than an hour, wait an hour before refreshing
+        if (expiration == null || expiration.isAfter(oneHourFromNow)) {
+            return oneHourFromNow;
+        }
+
+        // If expiration time is within 15 minutes (or in the past), wait 15 minutes and warn the customer that they'll be using
+        // expired credentials.
+        Instant fifteenMinutesFromNow = now.plus(15, MINUTES);
+        if (expiration.isBefore(fifteenMinutesFromNow)) {
+            log.warn(() -> "IMDS credential expiration has been extended due to an IMDS availability outage. A refresh"
+                           + " of these credentials will be attempted again in 15 minutes.");
```

Comment:
Should we log warning here? I wonder if this is necessary because there's nothing users can do when they notice the log